### ISSUE TITLE
Guild incorrectly parsing numeric str flags

### ISSUE THREAD

**Issue by FelixAbrahamsson** [guildai/guildai#303]
I have an operation `test.py` like this
```
import argparse


if __name__ == '__main__':

    parser = argparse.ArgumentParser()
    parser.add_argument('--some_flag', type=str, default='')
    args = parser.parse_args()
    print(args.some_flag)
```
If I run `guild run test some_flag=00500` then `some_flag` is parsed as `500` despite being explicitly typed as `str`. Running `python test.py --some_flag=00500` correctly parses it as the string `'00500'`.

**Comment by smao-astro** [comment]
Maybe try `guild run test some_flag='00500'`

**Comment by FelixAbrahamsson** [comment]
I tried that and it does not work either.

**Comment by gar1t** [comment]
Apologies for the late reply here! The single level of quote on bash in @smao-astro would be used to quote white space - the quotes themselves never make it to the program. You need something like this:

    $ guild run test some_flag="'00500'"

That said, this looks like a bug in Guild. Guild should use `type='str'` in the arg parser to infer that type and not attempt to parse the string as YAML (its default behavior).

I'll create a resolution doc to look into this. In the meantime, use the quoting syntax above and Guild should correctly parse the arg as a string.

**Comment by gar1t** [comment]
This is all confirmed and documented here:

https://github.com/guildai/issue-resolution/tree/master/303-guild-incorrectly-parsing-numeric-str-flags

I'll look into the issue and hopefully have a fix for the next release!

**Comment by gar1t** [comment]
A fix to this is committed in master. Thanks for the report and sorry for the inconvenience!

**Comment by gar1t** [comment]
I configured that this is fixed in 0.7.4. This can be verified [here](https://github.com/guildai/issue-resolution/blob/master/303-guild-incorrectly-parsing-numeric-str-flags/FIX.md).
